model/user.go

```diff
@@ -1,6 +1,78 @@
 package model
 
+import (
+	"errors"
+	"log"
+
+	helper "web/helpers"
+
+	"gorm.io/gorm"
+)
+
+var DB *gorm.DB
+
 type User struct {
-	UserID string `json:"user_id"`
-	Name   string `json:"name"`
+	gorm.Model
+	Email    string `gorm:"size64, index"`
+	Username string `gorm:"size64, index"`
+	Password string `gorm:"size255"`
+}
+
+func CheckUserExistance(email string) bool {
```

Comment:
All of this is in a model folder, but actually they are repository actions with the DB, so I would moved it under repository folder as there is also one of it - get_users.go